sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -154,30 +155,54 @@ function getRequestBody(body?: unknown, contentType: string = ""): RequestBody {
   }
 }
 
-function isFormData(body: unknown): body is FormDataMap {
-  return body instanceof Object && Object.keys(body).length > 0;
+/**
+ * Union of possible input types for multipart/form-data values that are accepted by RLCs.
+ * This extends the default FormDataValue type to include Uint8Array, which is accepted as an input by RLCs.
+ */
+type RLCFormDataValue = FormDataValue | Uint8Array;
```

Comment:
The name with prefix `RLC` looks wired to me.

As the core perspective they may not care about RLC or Modular or others. They may care about the generic types which they could support to handle. 

On the other hand it would couple the RLC designs with core implementations but it's un-necessary. RLC could adjust its types and build helpers to adopt to core types.

Reply:
That's fair. I was not 100% on the name of this type either, and am more than happy to rename it if you have a suggestion. I just couldn't think of a good alternative at the time. It's not exported or anything, it's just used as a helper to save a bit of typing further down in the code, so I didn't give it much thought.

In regard to adjusting RLC types and removing coupling, I agree in principle we should avoid coupling wherever we can, but the fact of the matter is that we're already coupled to an extent. We need to handle the `Uint8Array` case here, even though that's not the shape that core-rest-pipeline expects -- [we've been handling it here for a long time already](https://github.com/Azure/azure-sdk-for-js/pull/18753). Removing that handling would be a breaking change.

Reply:
Since it is not public I think I'm fine with the RLC prefix, it helps making it clear whats going on

Reply:
Agreed that it's not a good public name, but I'm ok leaving it as long as we revisit in the case that it becomes public later